components/form/demo/form-dialog-demo.js

```diff
@@ -0,0 +1,85 @@
+
+import '../../button/button.js';
+import '../../button/floating-buttons.js';
+import '../../dialog/dialog.js';
+import '../../inputs/input-text.js';
+import '../../validation/validation-custom.js';
+import '../form.js';
+import './form-panel-demo.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { inputStyles } from '../../inputs/input-styles.js';
+import { selectStyles } from '../../inputs/input-select-styles.js';
+
+class FormDialogDemo extends LitElement {
+
+	static get styles() {
+		return [inputStyles, selectStyles, css`
+
+			.d2l-form-dialog-demo-container {
+				margin-bottom: 10px;
+			}
+		`];
+	}
+
+	render() {
+		return html`
+			<d2l-form id="dialog-main-form">
+				<div class="d2l-form-dialog-demo-container">
+					<label>Email<input class="d2l-input" name="email" type="email"></label>
+				</div>
+				<div class="d2l-form-dialog-demo-container">
+					<d2l-validation-custom for="password-dialog" @d2l-validation-custom-validate=${this._validatePassword} failure-text="Expected hunter2" ></d2l-validation-custom>
+					<label>Password<input class="d2l-input" id="password-dialog" name="password" required type="password"></label>
+				</div>
+				<d2l-dialog id="dialog" title-text="My Favorites">
+					<d2l-form id="dialog-secondary-form" async-submit no-nesting @d2l-form-submit=${this._onDialogSubmit}>
```

Comment:
What are your thoughts about a use case similar to this, where you have a bunch of content that's hidden (like in a closed dialog) that when hidden you don't want to participate in the form but when displayed you do? 

Could be as simple as them manually adding/removing the `no-nesting` attribute based on the visibility for now. But it would be nice for the nested `d2l-form` to not actually be necessary at all -- just have some hidden content that only participates when it's visible.

Reply:
Something like this would need to be approached carefully since I don't want to deviate too far from how native forms work where hidden inputs still get submitted.

The best solution that comes to mind is to use a `MutationObserver` and then conditional rendering can be used to include or exclude sections. This might not play well if we want to add Polymer support due to how `dom-if` and stamping works but I'm not sure how concerned we'd be about that.

Reply:
Good point about native forms, maybe I'm overthinking things. 👍